File: 프로그래머스/2/42747. H－Index/H－Index.js
function solution(citations) {
    let h = Math.max(...citations);
    let hIndex = 0;
    
    for(let i=1;i<=h;i++){
        let cntcit =0;
        for(let j=0;j<citations.length;j++){
            if(citations[j]>=i) cntcit++;
        }
        if(cntcit>=i){
            hIndex = i;
        }else break;
    }
    return hIndex;
}

File: 프로그래머스/2/42747. H－Index/H－Index.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    sort(citations.begin(), citations.end(), greater<int>());

    int h = 0;
    for (int i = 0; i < (int)citations.size(); ++i) {
        int paperCount = i + 1;
        int citeCount  = citations[i];

        if (citeCount >= paperCount) {
            h = paperCount;
        } else {
            break;
        }
    }
    return h;
}
